From: Sophie Brun <sophie@offensive-security.com>
Date: Wed, 2 Oct 2019 17:52:52 +0200
Subject: Switch to Python 3

Last-Update: 2019-10-02
---
 src/bff/bffxtractor.py | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/src/bff/bffxtractor.py b/src/bff/bffxtractor.py
index 80ab1af..9d2862b 100755
--- a/src/bff/bffxtractor.py
+++ b/src/bff/bffxtractor.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # A hacky extraction utility for extracting the contents of BFF volume entries.
 # It can't parse a BFF file itself, but expects the BFF volume entry to already 
 # be extracted to a file; it then extracts the original file from the volume entry 
@@ -40,7 +40,7 @@ script_path = os.path.dirname(os.path.realpath(__file__))
 try:
 	fd = open(sys.argv[1], 'rb')
 except:
-	print "Usage: %s <BFF volume entry file>" % sys.argv[0]
+	print("Usage: %s <BFF volume entry file>" % sys.argv[0])
 	sys.exit(1)
 
 header = fd.read(HEADER_SIZE)
@@ -49,7 +49,7 @@ magic = struct.unpack("<H", header[2:4])[0]
 file_size = struct.unpack("<L", header[56:60])[0]
 
 if magic not in MAGICS:
-	print "Unrecognized magic bytes! Quitting."
+	print("Unrecognized magic bytes! Quitting.")
 	sys.exit(1)
 
 filename = ''
@@ -64,10 +64,10 @@ filename_len = len(filename)
 offset = HEADER_SIZE + POST_HEADER_SIZE + filename_len + (8 - (filename_len % 8))
 
 if '..' in filename:
-	print "Dangerous file path '%s'! Quitting." % filename
+	print("Dangerous file path '%s'! Quitting." % filename)
 	sys.exit(1)
 
-print "Extracting '%s'" % filename
+print("Extracting '%s'" % filename)
 _mkdir('./' + os.path.dirname(filename))
 if file_size:
 	fd.seek(offset)
@@ -75,7 +75,7 @@ if file_size:
 	fd.close()
 
 	if len(file_data) != file_size:
-		print "Warning: EOF encountered before the expected file size was reached!"
+		print("Warning: EOF encountered before the expected file size was reached!")
 
 	if magic == HUFFMAN_MAGIC:
 		fpout = open(filename + '.packed', 'wb')
